### PR TITLE
Feature/zke central locking

### PR DESCRIPTION
**Pull request by piersholt**
I'll add some comments to the code, but quick overview:

- The handlers are a bit of a departure. I've split out doors and central locking into their own functions for ease. The event data is now `uint8_t` rather than `unsigned char`.

- I've opted to add central locking/doors status to both IBus and Handler contexts. I'm not yet sure what the comfort locks implementation will look like, so I figured I'd add them both for now and remove anything that might ultimately be redundant down the track.

- I've intentionally left the existing doors/locks handling for the moment. I'll remove that on a later branch.

- As with the previous branch, this is just groundwork. Some refactoring to come.

- Builds without warning/error 👍

### CODE REVIEW

File: firmware/application/handler.c [hunk 1]
```diff
@@ -180,6 +180,18 @@ void HandlerInit(BC127_t *bt, IBus_t *ibus)
         &HandlerIBusGMIdentResponse,
         &Context
     );
+
```
Comment (piersholt) [inline firmware/application/handler.c:183]: I'll remove these errant new lines later!

File: firmware/application/handler.c [hunk 2]
```diff
@@ -937,6 +949,94 @@ void HandlerIBusFirstMessageReceived(void *ctx, unsigned char *pkt)
...
+ * HandlerIBusGMCentralLockingStatus()
+ *     Description:
+ *         Track the General Module's central locking state.
+ *     Params:
+ *         void *ctx - The context provided at registration
+ *         uint8_t *centralLockingStatus - Any event data
+ *     Returns:
+ *         void
+ */
+void HandlerIBusGMCentralLockingStatus(void *ctx, uint8_t *centralLockingStatus)
```
Comment (piersholt) [inline firmware/application/handler.c:962]: Note `uint8_t`!

File: firmware/application/lib/ibus.h [hunk 3]
```diff
@@ -122,6 +122,30 @@
...
+// 0x7A: doors status
+#define IBUS_CMD_GM_DOORS 0x0F
+#define IBUS_CMD_GM_DOORS_DRIVER 0x01
+#define IBUS_CMD_GM_DOORS_PASSENGER 0x02
+#define IBUS_CMD_GM_DOORS_REAR_RIGHT 0x04
+#define IBUS_CMD_GM_DOORS_REAR_LEFT 0x08
+
+// I appreciate this looks redundant, but I want to distinguish
+// between IBus data and our data structures
+#define IBUS_DOORS_DRIVER 0x1
```
Comment (piersholt) [inline firmware/application/lib/ibus.h:134]: Not sure if the toolkit supports binary literals?

**Review by tedsalmon — APPROVED** [review]
Looks great, thanks a bunch! :smile:

**Pull request merged**